src/include/interval_set.h

```diff
@@ -123,32 +123,32 @@ class interval_set {
         }
 
         // Dereference this iterator to get a pair.
-        std::pair < T, T > operator*() const {
+        auto& operator*() const {
```

Comment:
@jtlayton this fixes the problem you noted in https://github.com/ceph/ceph/pull/31817#discussion_r356012788

Reply:
Are you sure about that? I think the problem is in the iterator subclass, not const_iterator.

Reply:
i.e. iterator::operator* returns a reference to a pair; map::iterator returns a value.

Reply:
Right, I linked to the wrong line. It's here actually: https://github.com/ceph/ceph/pull/32185/files#diff-e9ca413caeab2ab2288ab1706f2877afR59

The issue is that the method return type is wrong by definition but since nothing was calling it, no errors were raised during compilation (until Jeff tried using it).

Reply:
Exactly. I just saw the line you linked and didn’t realize you just changed everything to auto.

Reply:
agreed. the `iterator` fix addresses the issue. but the crux of the problem is that the divergency between `std::pair < T, T >` and `std::pair <const T, T >`. the latter is `Map::reference`.